plugins/woocommerce/src/StoreApi/StoreApi.php

```diff
@@ -35,6 +35,24 @@ function() {
 			},
 			11
 		);
+
+		add_action(
+			'woocommerce_blocks_pre_get_routes_from_namespace',
+			function( $routes, $namespace, $context ) {
+				if ( 'wc/store/v1' !== $namespace ) {
+					return array();
+				}
+
+				$routes = array_merge(
+					$routes,
+					self::container()->get( RoutesController::class )->get_all_routes( 'v1' )
+				);
+
+				return $routes;
+			},
+			10,
+			3
+		);
```

Comment:
It's not explained anywhere why is this needed.

Reply:
My bad, it's buried in the wall of text in description. So this adds a hook 
 [in `woocommerce_blocks_pre_get_routes_from_namespace` ](https://github.com/woocommerce/woocommerce/pull/44080/files/bc5d51b0208f8609f2a180fa82b63f1b15ac3b56#diff-aab09b3f15d65e08d6da3daf65c24daa0564dee14773891f401ccf2683d761b5R474) and makes use of the opportunity to provide a pre-built routes array without loading the REST API. If we don't hook here, than the corresponding `apply_filter` will end up loading the REST API to build store API routes.